pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -215,42 +234,57 @@ func getPortMap(pod *v1.Pod) map[string]uint32 {
 	return pmap
 }
 
-// deleteIP returns true if the pod and ip are really deleted.
-func (pc *PodCache) deleteIP(ip string, podKey types.NamespacedName) bool {
+// deleteIPs returns true if the pod and ip are really deleted.
+func (pc *PodCache) deleteIPs(ips []string, podKey types.NamespacedName) bool {
 	pc.Lock()
 	defer pc.Unlock()
-	if pc.podsByIP[ip] == podKey {
-		delete(pc.podsByIP, ip)
-		delete(pc.IPByPods, podKey)
-		return true
+	deleteCounter := 0
+	for _, itemIP := range ips {
+		if pc.podsByIP[itemIP] == podKey {
+			delete(pc.podsByIP, itemIP)
+			delete(pc.IPByPods, podKey)
+			deleteCounter++
+		}
 	}
-	return false
+	return deleteCounter == len(ips)
 }
 
-func (pc *PodCache) update(ip string, key types.NamespacedName) {
+func (pc *PodCache) update(ips []string, key types.NamespacedName) {
 	pc.Lock()
 	// if the pod has been cached, return
-	if key == pc.podsByIP[ip] {
+	var isCachedAll bool
+	for _, itemIP := range ips {
+		if key == pc.podsByIP[itemIP] {
+			isCachedAll = true
```

Comment:
This logic is wrong, we end up only checking the last IP

Reply:
Hi @howardjohn, understand, should check all IPs passed from the parameter at the same time

Reply:
done